jar.go

```diff
@@ -396,6 +396,34 @@ func (j *Jar) deleteExpired(now time.Time) {
 	}
 }
 
+// RemoveAll removes any cookies from the jar that were set for the given host.
```

Comment:
This comment isn't technically correct. It removes all cookies that were set
for any host with the same public suffix as host, which is quite
a different thing.

Is this really the behaviour we want?